frontend/src/app/main/data/workspace/selection.cljs

```diff
@@ -47,51 +47,58 @@
       (assoc-in state [:workspace-local :selrect] selrect))))
 
 (defn handle-area-selection
-  [preserve?]
-  (letfn [(data->selrect [data]
-            (let [start (:start data)
-                  stop (:stop data)
-                  start-x (min (:x start) (:x stop))
-                  start-y (min (:y start) (:y stop))
-                  end-x (max (:x start) (:x stop))
-                  end-y (max (:y start) (:y stop))]
-              {:type :rect
-               :x start-x
-               :y start-y
-               :width (mth/abs (- end-x start-x))
-               :height (mth/abs (- end-y start-y))}))]
-    (ptk/reify ::handle-area-selection
-      ptk/WatchEvent
-      (watch [_ state stream]
-        (let [zoom (get-in state [:workspace-local :zoom] 1)
-              stop? (fn [event] (or (dwc/interrupt? event) (ms/mouse-up? event)))
-              stoper (->> stream (rx/filter stop?))
-
-              calculate-selrect
-              (fn [data pos]
-                (if data
-                  (assoc data :stop pos)
-                  {:start pos :stop pos}))
-
-              selrect-stream
-              (->> ms/mouse-position
-                   (rx/scan calculate-selrect nil)
-                   (rx/map data->selrect)
-                   (rx/filter #(or (> (:width %) (/ 10 zoom))
-                                   (> (:height %) (/ 10 zoom))))
-                   (rx/take-until stoper))]
-          (rx/concat
-           (if preserve?
-             (rx/empty)
-             (rx/of (deselect-all)))
-
-           (rx/merge
-            (->> selrect-stream (rx/map update-selrect))
-            (->> selrect-stream
-                 (rx/debounce 50)
-                 (rx/map #(select-shapes-by-current-selrect preserve?))))
-
-           (rx/of (update-selrect nil))))))))
+  [preserve? ignore-groups?]
+  (ptk/reify ::handle-area-selection
+    ptk/WatchEvent
+    (watch [_ state stream]
+      (let [zoom (get-in state [:workspace-local :zoom] 1)
+            stop? (fn [event] (or (dwc/interrupt? event) (ms/mouse-up? event)))
+            stoper (->> stream (rx/filter stop?))
+
+            init-selrect
+            {:type :rect
+             :x (:x @ms/mouse-position)
+             :y (:y @ms/mouse-position)
+             :width 0
+             :height 0}
+
+            calculate-selrect
+            (fn [selrect [delta space?]]
+              (if space?
+                (-> selrect
+                    (update :x + (:x delta))
+                    (update :y + (:y delta)))
+
+                (-> selrect
+                    (update :width + (:x delta))
+                    (update :height + (:y delta)))))
+
+            selrect-stream
+            (->> ms/mouse-position
+                 (rx/take-until stoper)
```

Comment:
The same here, stopper should be on the end of stream.

Reply:
ok